### ISSUE TITLE
[Android] TextInput ` keyboardType={'numeric'}` broke `secureTextEntry`

### ISSUE THREAD

**Issue by qbig** [facebook/react-native#4090]
```
  <TextInput
            secureTextEntry={true}
            keyboardType={'numeric'}
            style={styles.emptyText} />
```

if I use `keyboardType={'numeric'}` instead of keyboardType={'default'}, the text would no long be hidden as bullet while keying in. Is this an intended behaviour ?

**Comment by nicklockwood** [comment]
No, it's definitely not intended.

**Comment by nicklockwood** [comment]
cc @mkonicek

**Comment by mkonicek** [comment]
Looks like a bug although doesn't seem like a huge risk since anyone using `numeric` with `secureTextEntry` would probably notice this immediately.

@andreicoman11 I assume we (want to) support `keyboardType={'numeric'}` on Android?

@qbig Would you be up for sending a pull request to fix this? It would likely be in `ReactTextViewManager` or `ReactTextView`.

**Comment by qbig** [comment]
@mkonicek Yup. I will give it a go.

**Comment by qbig** [comment]
@andreicoman11 Anything I need to take note on this?

**Comment by andreicoman11** [comment]
The bug is probably in ReactTextInputManager, we might be clearing some flags with the numerical keyboard flag that we shouldn't have cleared.

**Comment by qbig** [comment]
hmmm. 
```
  @ReactProp(name = "password", defaultBoolean = false)
  public void setPassword(ReactEditText view, boolean password) {
    updateStagedInputTypeFlag(
        view,
        password ? 0 : InputType.TYPE_TEXT_VARIATION_PASSWORD,  //0x00000080
        password ? InputType.TYPE_TEXT_VARIATION_PASSWORD : 0);
  }
//....
 @ReactProp(name = "keyboardType")
  public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
    int flagsToSet = 0;
    if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
      flagsToSet = InputType.TYPE_CLASS_NUMBER;  
    } else if (KEYBOARD_TYPE_EMAIL_ADDRESS.equalsIgnoreCase(keyboardType)) {
      flagsToSet = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS; 
    }
    updateStagedInputTypeFlag(
        view,
        InputType.TYPE_CLASS_NUMBER | InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS,
        flagsToSet);
  }
```

These code seems pretty harmless. But what Android expects for numeric + password is actually InputType.TYPE_CLASS_NUMBER together with an InputType.TYPE_NUMBER_VARIATION_PASSWORD flag.

```
    /**
     * Variation of {@link #TYPE_CLASS_NUMBER}: entering a numeric password.
     * This was added in {@link android.os.Build.VERSION_CODES#HONEYCOMB}.  An
     * IME must target this API version or later to see this input type; if it
     * doesn't, a request for this type will be dropped when passed
     * through {@link android.view.inputmethod.EditorInfo#makeCompatible(int)
     * EditorInfo.makeCompatible(int)}.
     */
    public static final int TYPE_NUMBER_VARIATION_PASSWORD = 0x00000010;
```

So to certain extend, current implementation behave consistently with native Android code, as discussed in the following SO questions as well. 
[SO discussion here:)](http://stackoverflow.com/questions/9880516/how-do-i-specify-edittext-inputtype-as-number-and-password)

More like TYPE_NUMBER_VARIATION_PASSWORD is missing. So not really a 'bug' here, I would say.

In other words, the code below doesn't work either.
```
//native Android code
<EditText
    android:inputType="number"
    android:password="true"
    ....
</EditText>
```

So what should we do here ? @andreicoman11  @mkonicek

**Comment by andreicoman11** [comment]
We should add support for `InputType.TYPE_NUMBER_VARIATION_PASSWORD` then. In `onAfterUpdateTransaction` we should check if we have both password and numeric keyboard set or unset, and update the flags accordingly.

**Comment by mkonicek** [comment]
I'll close this as Andrei answered. Please send a PR to fix this.

**Comment by andreicoman11** [comment]
Fixed in https://github.com/facebook/react-native/commit/cc449b394e9cd3300343016a5d6a3c924823cd2b

**Comment by nihgwu** [comment]
now keyboardType={'email-address'}` and keyboardType={'phone-pad'}` broke `secureTextEntry` too,  
seems those two types were added later?
